CONTRIBUTING.md

```diff
@@ -68,38 +69,97 @@ Please see `./ci/features`.
 To test the usage of the Conjur Service Broker within a CF deployment, you can
 follow the demo scripts in the [Cloud Foundry demo repo](https://github.com/conjurinc/cloudfoundry-conjur-demo).
 
-To run the test suite on your local machine:
+#### Running the Dev Environment
+To test your changes within a running instance of Cloud Foundry and Conjur, run:
+
+```shell script
+./ci/start_dev_environment
+```
+
+This starts Conjur and Cloud Foundry containers, and provides terminal 
+access to the Cloud Foundry container. You do not need to restart the container
+after you make changes to the project.
+
+To run tests within the development environment, run the following command from
+the `ci` directory:
+```shell script
+cucumber \
+    --format pretty \
+    --format junit \
+    --out ./features/reports \
+    --tags 'not @integration'
 ```
-$ ./package.sh   # Create the conjur-env binary in the vendor dir and a ZIP of the project contents
-$ ./test.sh      # Run the test suite
+
+
+#### Setup
+To run the test suite on your local machine, first run:
+```shell script
+./package.sh
 ```
+This will create the conjur-env binary in the vendor dir and a ZIP of the project contents.
 
 #### Integration Testing
 
 To run the buildpack integration tests, the test script needs to be given the API endpoint and admin credentials
 for a CloudFoundry installation. These are provided as environment variables to the script:
 
-```sh-session
-$ export CF_API_ENDPOINT=https://api.sys.cloudfoundry.net
-$ CF_ADMIN_PASSWORD=... ./test.sh
+```shell script
+export CF_API_ENDPOINT=https://api.sys.cloudfoundry.net
+CF_ADMIN_PASSWORD=... ./test.sh
 ```
 
 These variables may also be provided using [Summon](https://cyberark.github.io/summon/) by updating the `secrets.yml`
 file as needed and running:
 
+```shell script
+summon ./ci/test_integration_remote
 ```
-$ summon ./test.sh
+This requires access to privileged summon credentials.
+
+Alternatively, you can run the set of features marked with `not @integration`, 
+which are the subset of tests not dependent on a remote PCF instance or privileged credentials.
+Run:
+
+```shell script
+./ci/test_integration
+```
+
+#### Unit Testing
+Unit tests are comprised of two categories:
+- Unit tests, linting, and code coverage for `conjur-env` Golang module
+- Unit tests for `lib/0001_retrieve-secrets.sh`
+
+To run all tests for the `conjur-env` Golang module *and* for 
+`lib/0001_retrieve-secrets.sh`, you can run:
+
+```shell script
+./ci/test_unit
 ```
 
-#### Testing `0001_retrieve-secrets.sh` 
-To test `0001_retrieve-secrets.sh` locally:
+To run all tests for _only_ the `conjur-env` Golang module, run:
 
 ```shell script
-$ ./ci/test-retrieve-secrets/start
+./ci/test_conjur-env
+```
+
+To run all tests for _only_ `0001_retrieve-secrets.sh`, run:
+
+```shell script
+./ci/test-retrieve-secrets/start
```

Comment:
I'm thinking `run` is a more appropriate name for this script i.e. `./ci/test-retrieve-secrets/run`

Reply:
I think the two words are fairly similar in the way we would use them, but `start`, `test`, and `stop` are the words we use for test frameworks in other projects, so I think we should stick to `start`.